codalab/rest/users.py

```diff
@@ -140,13 +140,44 @@ def fetch_users():
         filter[user_name]=name1,name2,...
         filter[email]=email1,email2,...
 
-    Fetches all users that match any of these usernames or emails.
+    Query parameters:
+
+    - `keywords`: Search keyword. May be provided multiples times for multiple
+    keywords.
+    Examples of other special keyword forms:
+    - `name=<name>            ` : More targeted search of using metadata fields.
+    - `date_joined=.sort             ` : Sort by a particular field.
+    - `date_joined=.sort-            ` : Sort by a particular field in reverse.
+    - `.count                 ` : Count the number of users.
+    - `.limit=10              ` : Limit the number of results to the top 10.
     """
     # Combine username and email filters
     usernames = set(request.query.get('filter[user_name]', '').split(','))
     usernames |= set(request.query.get('filter[email]', '').split(','))
     usernames.discard('')  # str.split(',') will return '' on empty strings
-    users = local.model.get_users(usernames=(usernames or None))
+
+    keywords = query_get_list('keywords')
+    if usernames == None and keywords == None:
+        abort(
+            http.client.BAD_REQUEST, "Request must include 'keywords' query parameter or usernames"
+        )
+
+    if request.user.user_id != local.model.root_user_id:
+        for key in keywords:
+            if any(
+                unaccessed_field in key for unaccessed_field in ['time', 'disk', 'active', 'email']
```

Comment:
Why do we restrict only these fields? Did we already discuss which fields shouldn't be accessible? I'm wondering if other fields such as `parallel_run_quota` should also be inaccessible (alternatively, we could create a whitelist of allowed fields instead of a blacklist of disallowed fields)

cc @percyliang